TU20Bot/Commands/NameMatch.cs

```diff
@@ -0,0 +1,154 @@
+                            noLastNameMsg += $"`{lastName}` of `{firstName} {lastName}` matches " +
+                                             $"with `{origNames[i, 1]}` of `{origNames[i, 0]} {origNames[i, 1]}`\n";
+                            Console.WriteLine($"`{lastName}` of `{firstName} {lastName}` matches " +
+                                              $"with `{origNames[i, 1]}` of `{origNames[i, 0]} {origNames[i, 1]}`");
+                        }
+                    }
+                }
+            }
+
+            // For adding all the messages to send to the user
+            discordMessage += noLastNameMsg + noFirstNameMsg;
+            // For reporting if the user doesn't exist in the server (no last name match)
+            for (int i = 0; i < origNames.GetLength(0); i++) {
+                if (!discordMessage.Contains(origNames[i, 1])) {
+                    notInGuildMsg +=
+                        $"{origNames[i, 0]} {origNames[i, 1]} did not have a last name match therefore is not in the server\n";
+                }
+            }
+
+            discordMessage += notInGuildMsg;
+            await sendSplitMessage(discordMessage, "\n");
+        }
+
+        private bool compare(string name, string matchName) {
+            return name.Replace(" ", "").Contains(matchName.Replace(" ", ""));
+        }
+
+        private async Task sendSplitMessage(string mainMessage, string separator) {
+            // The max limit set by discord
+            int messageLimit = 2000;
+            int startIndex = 0;
+
+            if (mainMessage == null) {
+                return;
+            }
+
+            if (mainMessage.Length <= messageLimit) {
+                await ReplyAsync(mainMessage);
+                return;
+            }
+            // If the separator doesn't exist in the string then split it into simple parts
+            if (!mainMessage.Contains(separator)) {
+                do {
+                    await ReplyAsync(mainMessage.Substring(startIndex, messageLimit));
+                    startIndex = messageLimit;
+                } while (startIndex+messageLimit <= mainMessage.Length);
+                await ReplyAsync(mainMessage.Substring(startIndex));
+                return;
+            }
+
+            do {
+                // Get the substring of the specified length
+                string subString = mainMessage.Substring(startIndex, messageLimit + 1);
+                // Get the last index of new line in the substring of specified length
+                int endIndex = subString.LastIndexOf(separator);
+                // Send the message from the start index to last index of new line character
+                await ReplyAsync(mainMessage.Substring(startIndex, endIndex));
+                // set the start index to end index, +1 to skip over \n character
+                startIndex += endIndex + 1;
+            } while (startIndex + messageLimit <= mainMessage.Length);
```

Comment:
If you are sending separate messages based on a singular delimiter, you can use `string#Split(Char[])` on `mainMessage` and iterate through the array and send each message.

To further expand the solution, if you are expecting that you will be creating a buffer of multiple messages, you can also use a `List<string>` and iterate through it when you are ready to send the messages. This would remove the need to creating a new array every time a message queue needs to be sent.